bindings/python/pinocchio/shortcuts.py

```diff
@@ -84,36 +111,84 @@ def createDatas(*models):
 
 
 def buildModelsFromSdf(
+    filename, *args, **kwargs
+) -> Tuple[pin.Model, pin.GeometryModel, pin.GeometryModel]:
+    """Parse the Sdf file given in input and return a Pinocchio Model and a list of Constraint Models, followed by corresponding GeometryModels of types specified by geometry_types, in the same order as listed.
+    Arguments:
+        - filename - name of the urdf file to load
+        - package_dirs - where the meshes of the urdf are located. (default - None)
+        - root_joint - Joint at the base of the model (default - None)
+        - root_joint_name - Name for the root_joint (default - "root_joint")
+        - root_link_name - Name of the body to use as root of the model (default - "")
+        - parent_guidance - Joint names which should be preferred for cases where two joints can qualify as parent. The other joint appears in the constraint_model. If empty, joint appearance order in .sdf is taken as default.
+        - verbose - print information of parsing (default - False)
+        - meshLoader - object used to load meshes (default - hpp::fcl::MeshLoader)
+        - geometry_types - Which geometry model to load. Can be pin.GeometryType.COLLISION, pin.GeometryType.VISUAL, both or None. (default - None])
+    Return:
+        Tuple of the models, in this order : model, collision model, and visual model.
+
+    Example:
+        model, collision_model, visual_model = buildModelsFromSdf(filename, root_joint, root_link_name, parent_guidance, verbose, meshLoader, geometry_types, root_joint_name="root_joint_name")
+    """
+
+    arg_keys = [
+        "package_dirs",
+        "root_joint",
+        "root_link_name",
+        "parent_guidance",
+        "verbose",
+        "meshLoader",
+        "geometry_types",
+    ]
+    # Handle the switch from old to new api
+    if len(args) >= 3:
+        if isinstance(args[2], str):
+            arg_keys = [
+                "package_dirs",
+                "root_joint",
+                "root_joint_name",
+                "root_link_name",
+                "parent_guidance",
+                "verbose",
+                "meshLoader",
+                "geometry_types",
+            ]
```

Comment:
Sadly, our check is not working here. In the standard function definition, root_link_name is the second argument and it's also a string :(

It's then not possible to overload this function...